### PR TITLE
Update azure.openai.chat connector

### PR DESCRIPTION
**Pull request by JayaniH**
# Description

Updates the `azure.openai.chat` connector to support the new API version with function calling capabilities and chat extensions.

Fixes https://github.com/wso2-enterprise/choreo/issues/24117

One line release note: 
- Update `azure.openai.chat` connector to support API version `2023-08-01-preview`

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Locally tested the existing and new functionality.

**Test Configuration**:
* Ballerina Version: 2201.7.2 (Swan Lake Update 7)
* Operating System: macOs Monterey
* Java SDK: 

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [x] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [ ] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [ ] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/azure.openai.chat/types.bal [hunk 1]
```diff
@@ -7,7 +10,7 @@ public type ConnectionConfig record {|
     # Provides Auth configurations needed when communicating with a remote HTTP endpoint.
     http:BearerTokenConfig|ApiKeysConfig auth;
     # The HTTP version understood by the client
-    http:HttpVersion httpVersion = http:HTTP_1_1;
+    http:HttpVersion httpVersion = http:HTTP_2_0;
```
Comment (LakshanSS) [inline openapi/azure.openai.chat/types.bal:13]: @JayaniH Did u test this connector locally?
Usually Azure connectors don't work with `HTTP_2_0`. We might have to change this as `HTTP_1_1`
Can you verify this?

Comment (JayaniH) [reply openapi/azure.openai.chat/types.bal:13]: When testing locally, the generated connector worked.

File: openapi/azure.openai.chat/types.bal [hunk 2]
```diff
@@ -66,80 +69,224 @@ public type ApiKeysConfig record {|
...
+    string recipient?;
     # The contents of the message
     string content;
+    # Whether the message ends the turn.
+    boolean end_turn?;
+    # The conversation context
+    Message_context? context?;
+};
+
+public type ChatCompletionsResponseCommon_usage record {
```
[the remark this answers is not in the document]
Comment (JayaniH) [reply openapi/azure.openai.chat/types.bal:88]: the spec is not written by us. however, we can relevant descriptions to the spec if needed.

File: openapi/azure.openai.chat/types.bal (hunk 1 above)
Comment (LakshanSS) [reply openapi/azure.openai.chat/types.bal:13]: If its tested locally, its fine

**Review by LakshanSS — APPROVED** [review]

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by LakshanSS — APPROVED** [review]

**Pull request merged**